### PR TITLE
Semantic tokens support

### LINKED ISSUE #112: Is SemanticToken support a possability?
### Discussed in https://github.com/mkslanc/ace-linters/discussions/104

<div type='discussions-op-text'>

<sup>Originally posted by **Ryex** March 17, 2024</sup>
I'm wondering if [Semantic Token](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens) support is on the roadmap?

I'm currently working of a project to create an editor and emulator for an toy MIPS variant used as a scripting language in a game. that variant has a working LSP built by the community that I managed to get working via wasm in a webworker. The LSP has beautiful  Semantic Token support but doesn't support highlighting. As such I've had to build my own highlighting rules. These rules can't capture the same context as the lsp though.
</div>

### PR DESCRIPTION
**Pull request by mkslanc**
Closes #112

### CODE REVIEW

File: packages/ace-linters/src/cdn-worker.ts [hunk 1]
```diff
@@ -132,5 +132,10 @@ function getServices(includeLinters: { [name in SupportedServices]: boolean | un
         return [];
     }
 
-    return allServices.filter(service => includeLinters[service.name]);
+    return allServices.filter(service => {
+        if (includeLinters.javascript) { // left for backward compatibility
+            return includeLinters["eslint"];
+        }
```
Comment (anijanyan) [inline packages/ace-linters/src/cdn-worker.ts:138]: seems to be wrong

File: packages/ace-linters/src/services/language-client.ts [hunk 2]
```diff
@@ -244,6 +258,19 @@ export class LanguageClient extends BaseService implements LanguageService {
         this.connection.sendRequest("initialize", message).then((params: lsp.InitializeResult) => {
             this.isInitialized = true;
             this.serviceCapabilities = params.capabilities as lsp.ServerCapabilities;
+            console.log(this.serviceCapabilities);
```
Comment (nightwing) [inline packages/ace-linters/src/services/language-client.ts:261]: forgotten log

File: packages/ace-linters/src/message-types.ts [hunk 3]
```diff
@@ -91,7 +92,7 @@ export class ChangeMessage extends BaseMessage {
 }
 
 export class DeltasMessage extends BaseMessage {
-    type: MessageType = MessageType.applyDelta;
+    type: MessageType.applyDelta = MessageType.applyDelta;
```
Comment (nightwing) [inline packages/ace-linters/src/message-types.ts:95]: Why is this not inferred automatically?

File: packages/ace-linters/src/services/language-client.ts (hunk 2 above)
Comment (mkslanc) [reply packages/ace-linters/src/services/language-client.ts:261]: good catch

File: packages/ace-linters/src/message-types.ts (hunk 3 above)
Comment (mkslanc) [reply packages/ace-linters/src/message-types.ts:95]: Not sure, seems limitation on TS side, without it there is no narrowing in switch-case blocks

File: packages/ace-linters/src/services/language-client.ts (hunk 2 above)
Comment (anijanyan) [inline packages/ace-linters/src/services/language-client.ts:261]: test code

**Pull request merged**